File: packages/ts/tracing/src/debug-mode.ts
let debugModeEnabled = false;

/** Accepts `"1"` (preferred) and `"true"` (backwards compatible). Case-sensitive. */
function isEnvFlagEnabled(name: string): boolean {
  const value = process.env[name];
  return value === "1" || value === "true";
}

export function enableDebugMode(): void {
  debugModeEnabled = true;
}

export function disableDebugMode(): void {
  debugModeEnabled = false;
}

export function isDebugModeEnabled(): boolean {
  return debugModeEnabled || isEnvFlagEnabled("LEMMA_DEBUG");
}

export function isDebugVerifyEnabled(): boolean {
  return isEnvFlagEnabled("LEMMA_DEBUG_VERIFY");
}

export function lemmaDebug(prefix: string, msg: string, data?: Record<string, unknown>): void {
  if (!isDebugModeEnabled()) return;
  if (data !== undefined) {
    console.log(`[LEMMA:${prefix}] ${msg}`, data);
  } else {
    console.log(`[LEMMA:${prefix}] ${msg}`);
  }
}

File: packages/py/tracing/src/uselemma_tracing/debug_mode.py
from __future__ import annotations

import os

_debug_mode_enabled = False


def _is_env_flag_enabled(name: str) -> bool:
    """Accepts ``"1"`` (preferred) and ``"true"`` (backwards compatible). Case-sensitive."""
    value = os.environ.get(name)
    return value == "1" or value == "true"


def enable_debug_mode() -> None:
    global _debug_mode_enabled
    _debug_mode_enabled = True


def disable_debug_mode() -> None:
    global _debug_mode_enabled
    _debug_mode_enabled = False


def is_debug_mode_enabled() -> bool:
    return _debug_mode_enabled or _is_env_flag_enabled("LEMMA_DEBUG")


def is_debug_verify_enabled() -> bool:
    return _is_env_flag_enabled("LEMMA_DEBUG_VERIFY")


def _lemma_debug(prefix: str, msg: str, **data: object) -> None:
    if not is_debug_mode_enabled():
        return
    if data:
        print(f"[LEMMA:{prefix}] {msg}", data)
    else:
        print(f"[LEMMA:{prefix}] {msg}")
